docs/notebooks/trust_region.pct.py

```diff
@@ -0,0 +1,286 @@
+# We can use trust regions for Bayesian optimization in much the same way as we used EGO and EI in
+# the [introduction notebook](expected_improvement.ipynb). Since the setup is very similar to
+# that tutorial, we'll skip over most of the detail.
+
+# %%
+import trieste
+from trieste.objectives import Branin
+
+branin = Branin.objective
+search_space = Branin.search_space
+
+num_initial_data_points = 10
+initial_query_points = search_space.sample(num_initial_data_points)
+observer = trieste.objectives.utils.mk_observer(branin)
+initial_data = observer(initial_query_points)
+
+# %% [markdown]
+# As usual, we'll use Gaussian process regression to model the function. Note that we set the
+# likelihood variance to a small number because we are dealing with a noise-free problem.
+
+# %%
+from trieste.models.gpflow import GaussianProcessRegression, build_gpr
+
+
+def build_model():
+    gpflow_model = build_gpr(
+        initial_data, search_space, likelihood_variance=1e-7
+    )
+    return GaussianProcessRegression(gpflow_model)
+
+
+# %% [markdown]
+# ## Trust region `TREGO` acquisition rule
+#
+# First we show how to run Bayesian optimization with the `TREGO` algorithm. This is a trust region
+# algorithm that alternates between regular EGO steps and local steps within one trust region.
+#
+# ### Create `TREGO` rule and run optimization loop
+#
+# We can run the Bayesian optimization loop by defining a `BayesianOptimizer` and calling its
+# `optimize` method with the trust region rule. Once the optimization loop is complete, the
+# optimizer will return one new query point for every step in the loop; that's 5 points in total.
+
+# %%
+acq_rule = trieste.acquisition.rule.TrustRegion()
+bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+
+num_steps = 5
+result = bo.optimize(
+    num_steps, initial_data, build_model(), acq_rule, track_state=True
+)
+dataset = result.try_get_final_dataset()
+
+# %% [markdown]
+# ### Visualizing `TREGO` results
+#
+# Let's take a look at where we queried the observer, the original query points (crosses), new
+# query points (dots) and the optimum point found (purple dot), and where they lie with respect to
+# the contours of the Branin.
+
+# %%
+from trieste.experimental.plotting import plot_bo_points, plot_function_2d
+
+
+def plot_final_result(_dataset: trieste.data.Dataset) -> None:
+    arg_min_idx = tf.squeeze(tf.argmin(_dataset.observations, axis=0))
+    query_points = _dataset.query_points.numpy()
+    _, ax = plot_function_2d(
+        branin,
+        search_space.lower,
+        search_space.upper,
+        grid_density=40,
+        contour=True,
+    )
+
+    plot_bo_points(query_points, ax[0, 0], num_initial_data_points, arg_min_idx)
+
+
+plot_final_result(dataset)
+
+# %% [markdown]
+# We can also visualize the progress of the optimization by plotting the trust regions at each step.
+# The trust regions are shown as translucent boxes, with the current optimum point in each region
```

Comment:
For TREGO it is either the TR or the full search space. Change to

We can also visualize the progress of the optimization by plotting the acquisition space at each step.
This space is either the full search space or the trust region, depending on the step, and is shown as a translucent box, ...
?